include/LLVMSPIRVOpts.h

```diff
@@ -51,10 +51,12 @@ enum class VersionNumber : uint32_t {
   // Instruction
   SPIRV_1_0 = 0x00010000,
   SPIRV_1_1 = 0x00010100,
-  // TODO: populate this enum with the latest versions (up to 1.4) once
-  // translator get support of correponding features
+  SPIRV_1_2 = 0x00010200,
+  SPIRV_1_3 = 0x00010300,
```

Comment:
I'm not sure that this is critical, but there is some inconsistency: we do have SPIR-V 1.2 mentioned here, but we cannot limit maximum allowed SPIR-V version to it:

https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/b656125fb2ad0295f910630f1b3efcdab4f88dfe/tools/llvm-spirv/llvm-spirv.cpp#L101-L106

Reply:
It's not entirely clear to me what the behavior should be if a kernel uses these extended subgroup opcodes, which require SPIR-V 1.3, and the maximum SPIR-V version is set to SPIR-V 1.2 or earlier.

Should this:

* Fail gracefully?
* Emit SPIR-V 1.3 anyhow?
* Assert, because this is an invalid request?

As best I can tell right now, the maximum SPIR-V version is only used when deciding whether to emit some optional features that require a newer SPIR-V version.

Reply:
I think here the best thing would be "fail gracefully" or "assert, because this is an invalid request", see corresponding [docs](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/master/docs/SPIRVVersionsAndExtensionsHandling.rst#spir-v-generation-step):

> If the translator encounters something that cannot be represented by set of allowed SPIR-V versions (which might contain only one version), it does one of the following things:
>
>- ignores LLVM IR entity in the input file.
>
>   For example, dereferencable LLVM IR attribute can be ignored if it is not allowed to generate SPIR-V 1.1 and higher.
>
>- tries to represent LLVM IR entity with allowed instructions.
>
>   For example, `OpPtrEqual` can be used if SPIR-V 1.4 is not allowed and can be emulated via `OpConvertPtrToU` + `OpIEqual` sequence.
>
>- emits error if LLVM IR entity cannot be ignored and cannot be emulated using available instructions.
>
>   For example, if global constructors/destructors (represented by `@llvm.global_ctors`/`@llvm.global_dtors`) are present in a module then the translator should emit error if it cannot use SPIR-V 1.1 and higher where `Initializer` and `Finalizer` execution modes are described.

For now, I think I'm okay with some `TODO` about enabling proper version checks for new instructions